python/ml4ir/applications/ranking/model/ranking_model.py

```diff
@@ -314,17 +199,25 @@ def evaluate(
         for predictions_dict in test_dataset.map(_predict_fn).take(-1):
             predictions_df = pd.DataFrame(predictions_dict)
 
+            # Add necessary columns for power analysis
             clicked_records = predictions_df[predictions_df[self.feature_config.get_label("node_name")] == 1.0]
             clicked_records[RankingConstants.NEW_MRR] = 1.0 / clicked_records["new_rank"]
             clicked_records[RankingConstants.OLD_MRR] = 1.0 / clicked_records[self.feature_config.get_rank("node_name")]
 
-            running_stats_df = clicked_records.groupby('organization_id').agg({
-                    RankingConstants.NEW_MRR: ['mean', 'var'],
-                    RankingConstants.OLD_MRR: ['mean', 'var'],
+            # computing batch-wise mean and variance per metric
+            metric_stat_df_list = []
+            clicked_records.set_index("organization_id")
```

Comment:
This too would break for any dataset that does not contain this column. It feels we are trying to do quite many things together in single framework, but I can not offer an easy solution out of this currently.